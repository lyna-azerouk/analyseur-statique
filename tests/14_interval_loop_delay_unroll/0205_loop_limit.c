{
  int x;
  x = 0;
  while (rand(0,1)==0) {
    if (x < 10) x = x + 1;
  }
  print(x);
}
