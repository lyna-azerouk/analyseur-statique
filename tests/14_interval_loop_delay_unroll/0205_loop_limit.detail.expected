-- DEBUG: const 0 ~> [0;0]
-- DEBUG: const 0 ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:3.2-8: [ x in [0;0] ]
-- DEBUG: rand 0 1 ~> [0;1]
-- DEBUG: const 0 ~> [0;0]
-- DEBUG: [0;1] == [0;0] ~> [0;0], [0;0]
-- DEBUG: [0;1] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: [0;0] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: const 10 ~> [10;10]
-- DEBUG: [0;0] < [10;10] ~> [0;0], [10;10]
-- DEBUG: [0;0] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: [10;10] ⊓ [10;10] ~> [10;10]
-- DEBUG: is_bottom [10;10] ~> false
-- DEBUG: const 1 ~> [1;1]
-- DEBUG: [0;0] + [1;1] ~> [1;1]
-- DEBUG: is_bottom [1;1] ~> false
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:5.16-26: [ x in [1;1] ]
-- DEBUG: const 10 ~> [10;10]
-- DEBUG: [0;0] >= [10;10] ~> ⊥, ⊥
-- DEBUG: [0;0] ⊓ ⊥ ~> ⊥
-- DEBUG: is_bottom ⊥ ~> true
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:5.4-26: [ x in [1;1] ]
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:4.23-6.3: [ x in [1;1] ]
-- DEBUG: rand 0 1 ~> [0;1]
-- DEBUG: const 0 ~> [0;0]
-- DEBUG: [0;1] != [0;0] ~> [1;1], [0;0]
-- DEBUG: [0;1] ⊓ [1;1] ~> [1;1]
-- DEBUG: is_bottom [1;1] ~> false
-- DEBUG: [0;0] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: rand 0 1 ~> [0;1]
-- DEBUG: const 0 ~> [0;0]
-- DEBUG: [0;1] == [0;0] ~> [0;0], [0;0]
-- DEBUG: [0;1] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: [0;0] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: const 10 ~> [10;10]
-- DEBUG: [1;1] < [10;10] ~> [1;1], [10;10]
-- DEBUG: [1;1] ⊓ [1;1] ~> [1;1]
-- DEBUG: is_bottom [1;1] ~> false
-- DEBUG: [10;10] ⊓ [10;10] ~> [10;10]
-- DEBUG: is_bottom [10;10] ~> false
-- DEBUG: const 1 ~> [1;1]
-- DEBUG: [1;1] + [1;1] ~> [2;2]
-- DEBUG: is_bottom [2;2] ~> false
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:5.16-26: [ x in [2;2] ]
-- DEBUG: const 10 ~> [10;10]
-- DEBUG: [1;1] >= [10;10] ~> ⊥, ⊥
-- DEBUG: [1;1] ⊓ ⊥ ~> ⊥
-- DEBUG: is_bottom ⊥ ~> true
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:5.4-26: [ x in [2;2] ]
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:4.23-6.3: [ x in [2;2] ]
-- DEBUG: rand 0 1 ~> [0;1]
-- DEBUG: const 0 ~> [0;0]
-- DEBUG: [0;1] != [0;0] ~> [1;1], [0;0]
-- DEBUG: [0;1] ⊓ [1;1] ~> [1;1]
-- DEBUG: is_bottom [1;1] ~> false
-- DEBUG: [0;0] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: rand 0 1 ~> [0;1]
-- DEBUG: const 0 ~> [0;0]
-- DEBUG: [0;1] == [0;0] ~> [0;0], [0;0]
-- DEBUG: [0;1] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: [0;0] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: const 10 ~> [10;10]
-- DEBUG: [2;2] < [10;10] ~> [2;2], [10;10]
-- DEBUG: [2;2] ⊓ [2;2] ~> [2;2]
-- DEBUG: is_bottom [2;2] ~> false
-- DEBUG: [10;10] ⊓ [10;10] ~> [10;10]
-- DEBUG: is_bottom [10;10] ~> false
-- DEBUG: const 1 ~> [1;1]
-- DEBUG: [2;2] + [1;1] ~> [3;3]
-- DEBUG: is_bottom [3;3] ~> false
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:5.16-26: [ x in [3;3] ]
-- DEBUG: const 10 ~> [10;10]
-- DEBUG: [2;2] >= [10;10] ~> ⊥, ⊥
-- DEBUG: [2;2] ⊓ ⊥ ~> ⊥
-- DEBUG: is_bottom ⊥ ~> true
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:5.4-26: [ x in [3;3] ]
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:4.23-6.3: [ x in [3;3] ]
-- DEBUG: rand 0 1 ~> [0;1]
-- DEBUG: const 0 ~> [0;0]
-- DEBUG: [0;1] != [0;0] ~> [1;1], [0;0]
-- DEBUG: [0;1] ⊓ [1;1] ~> [1;1]
-- DEBUG: is_bottom [1;1] ~> false
-- DEBUG: [0;0] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: rand 0 1 ~> [0;1]
-- DEBUG: const 0 ~> [0;0]
-- DEBUG: [0;1] == [0;0] ~> [0;0], [0;0]
-- DEBUG: [0;1] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: [0;0] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: const 10 ~> [10;10]
-- DEBUG: [3;3] < [10;10] ~> [3;3], [10;10]
-- DEBUG: [3;3] ⊓ [3;3] ~> [3;3]
-- DEBUG: is_bottom [3;3] ~> false
-- DEBUG: [10;10] ⊓ [10;10] ~> [10;10]
-- DEBUG: is_bottom [10;10] ~> false
-- DEBUG: const 1 ~> [1;1]
-- DEBUG: [3;3] + [1;1] ~> [4;4]
-- DEBUG: is_bottom [4;4] ~> false
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:5.16-26: [ x in [4;4] ]
-- DEBUG: const 10 ~> [10;10]
-- DEBUG: [3;3] >= [10;10] ~> ⊥, ⊥
-- DEBUG: [3;3] ⊓ ⊥ ~> ⊥
-- DEBUG: is_bottom ⊥ ~> true
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:5.4-26: [ x in [4;4] ]
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:4.23-6.3: [ x in [4;4] ]
-- DEBUG: [3;3] ⊔ [4;4] ~> [3;4]
-- DEBUG: [3;4] ⊏ [3;3] ~> false
-- DEBUG: rand 0 1 ~> [0;1]
-- DEBUG: const 0 ~> [0;0]
-- DEBUG: [0;1] == [0;0] ~> [0;0], [0;0]
-- DEBUG: [0;1] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: [0;0] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: const 10 ~> [10;10]
-- DEBUG: [3;4] < [10;10] ~> [3;4], [10;10]
-- DEBUG: [3;4] ⊓ [3;4] ~> [3;4]
-- DEBUG: is_bottom [3;4] ~> false
-- DEBUG: [10;10] ⊓ [10;10] ~> [10;10]
-- DEBUG: is_bottom [10;10] ~> false
-- DEBUG: const 1 ~> [1;1]
-- DEBUG: [3;4] + [1;1] ~> [4;5]
-- DEBUG: is_bottom [4;5] ~> false
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:5.16-26: [ x in [4;5] ]
-- DEBUG: const 10 ~> [10;10]
-- DEBUG: [3;4] >= [10;10] ~> ⊥, ⊥
-- DEBUG: [3;4] ⊓ ⊥ ~> ⊥
-- DEBUG: is_bottom ⊥ ~> true
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:5.4-26: [ x in [4;5] ]
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:4.23-6.3: [ x in [4;5] ]
-- DEBUG: [3;3] ⊔ [4;5] ~> [3;5]
-- DEBUG: [3;5] ⊏ [3;4] ~> false
-- DEBUG: rand 0 1 ~> [0;1]
-- DEBUG: const 0 ~> [0;0]
-- DEBUG: [0;1] == [0;0] ~> [0;0], [0;0]
-- DEBUG: [0;1] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: [0;0] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: const 10 ~> [10;10]
-- DEBUG: [3;5] < [10;10] ~> [3;5], [10;10]
-- DEBUG: [3;5] ⊓ [3;5] ~> [3;5]
-- DEBUG: is_bottom [3;5] ~> false
-- DEBUG: [10;10] ⊓ [10;10] ~> [10;10]
-- DEBUG: is_bottom [10;10] ~> false
-- DEBUG: const 1 ~> [1;1]
-- DEBUG: [3;5] + [1;1] ~> [4;6]
-- DEBUG: is_bottom [4;6] ~> false
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:5.16-26: [ x in [4;6] ]
-- DEBUG: const 10 ~> [10;10]
-- DEBUG: [3;5] >= [10;10] ~> ⊥, ⊥
-- DEBUG: [3;5] ⊓ ⊥ ~> ⊥
-- DEBUG: is_bottom ⊥ ~> true
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:5.4-26: [ x in [4;6] ]
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:4.23-6.3: [ x in [4;6] ]
-- DEBUG: [3;3] ⊔ [4;6] ~> [3;6]
-- DEBUG: [3;6] ⊏ [3;5] ~> false
-- DEBUG: rand 0 1 ~> [0;1]
-- DEBUG: const 0 ~> [0;0]
-- DEBUG: [0;1] == [0;0] ~> [0;0], [0;0]
-- DEBUG: [0;1] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: [0;0] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: const 10 ~> [10;10]
-- DEBUG: [3;6] < [10;10] ~> [3;6], [10;10]
-- DEBUG: [3;6] ⊓ [3;6] ~> [3;6]
-- DEBUG: is_bottom [3;6] ~> false
-- DEBUG: [10;10] ⊓ [10;10] ~> [10;10]
-- DEBUG: is_bottom [10;10] ~> false
-- DEBUG: const 1 ~> [1;1]
-- DEBUG: [3;6] + [1;1] ~> [4;7]
-- DEBUG: is_bottom [4;7] ~> false
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:5.16-26: [ x in [4;7] ]
-- DEBUG: const 10 ~> [10;10]
-- DEBUG: [3;6] >= [10;10] ~> ⊥, ⊥
-- DEBUG: [3;6] ⊓ ⊥ ~> ⊥
-- DEBUG: is_bottom ⊥ ~> true
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:5.4-26: [ x in [4;7] ]
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:4.23-6.3: [ x in [4;7] ]
-- DEBUG: [3;3] ⊔ [4;7] ~> [3;7]
-- DEBUG: [3;7] ⊏ [3;6] ~> false
-- DEBUG: [3;6] ▿ [3;7] ~> [3;+∞]
-- DEBUG: rand 0 1 ~> [0;1]
-- DEBUG: const 0 ~> [0;0]
-- DEBUG: [0;1] == [0;0] ~> [0;0], [0;0]
-- DEBUG: [0;1] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: [0;0] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: const 10 ~> [10;10]
-- DEBUG: [3;+∞] < [10;10] ~> [3;9], [10;10]
-- DEBUG: [3;+∞] ⊓ [3;9] ~> [3;9]
-- DEBUG: is_bottom [3;9] ~> false
-- DEBUG: [10;10] ⊓ [10;10] ~> [10;10]
-- DEBUG: is_bottom [10;10] ~> false
-- DEBUG: const 1 ~> [1;1]
-- DEBUG: [3;9] + [1;1] ~> [4;10]
-- DEBUG: is_bottom [4;10] ~> false
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:5.16-26: [ x in [4;10] ]
-- DEBUG: const 10 ~> [10;10]
-- DEBUG: [3;+∞] >= [10;10] ~> [10;+∞], [10;10]
-- DEBUG: [3;+∞] ⊓ [10;+∞] ~> [10;+∞]
-- DEBUG: is_bottom [10;+∞] ~> false
-- DEBUG: [10;10] ⊓ [10;10] ~> [10;10]
-- DEBUG: is_bottom [10;10] ~> false
-- DEBUG: [4;10] ⊔ [10;+∞] ~> [4;+∞]
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:5.4-26: [ x in [4;+∞] ]
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:4.23-6.3: [ x in [4;+∞] ]
-- DEBUG: [3;3] ⊔ [4;+∞] ~> [3;+∞]
-- DEBUG: [3;+∞] ⊏ [3;+∞] ~> true
-- DEBUG: rand 0 1 ~> [0;1]
-- DEBUG: const 0 ~> [0;0]
-- DEBUG: [0;1] != [0;0] ~> [1;1], [0;0]
-- DEBUG: [0;1] ⊓ [1;1] ~> [1;1]
-- DEBUG: is_bottom [1;1] ~> false
-- DEBUG: [0;0] ⊓ [0;0] ~> [0;0]
-- DEBUG: is_bottom [0;0] ~> false
-- DEBUG: [3;+∞] ⊔ [2;2] ~> [2;+∞]
-- DEBUG: [2;+∞] ⊔ [1;1] ~> [1;+∞]
-- DEBUG: [1;+∞] ⊔ [0;0] ~> [0;+∞]
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:4.2-6.3: [ x in [0;+∞] ]
tests/14_interval_loop_delay_unroll/0205_loop_limit.c:7.2-11: [ x in [0;+∞] ]
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:7.2-11: [ x in [0;+∞] ]
stat trace: tests/14_interval_loop_delay_unroll/0205_loop_limit.c:1.0-8.1: [ ]
analysis ended
